### PR TITLE
Do not include unnecessary files in build

### PR DESCRIPTION
**Pull request by mikedn**
Some files have their contents ifdefed out because some CLR features are not present (and they'll probably never be) in CoreCLR. A few others are DAC and/or arch specific so they have been moved to the appropiate file lists.

### CODE REVIEW

File: src/utilcode/CMakeLists.txt [hunk 1]
```diff
@@ -25,7 +24,6 @@ set(UTILCODE_SOURCES
   stgpool.cpp
   stgpooli.cpp
   stgpoolreadonly.cpp
-  lazycow.cpp
```
Comment (jkotas) [inline src/utilcode/CMakeLists.txt]: This highlighted a mismatch between the old msbuild-based scripts and CMakefiles. This should be left in and FEATURE_LAZYCOW defined for x86 and arm instead (on Windows at least) - to match https://github.com/dotnet/coreclr/blob/master/clr.coreclr.props#L68.

**Comment by jkotas** [comment]
Looks good to me otherwise.

**Comment by mikedn** [comment]
Thanks, I restored lazycow.cpp (and enabled the feature) and cleaned up a few more files.

Do you know what's up with genericstackprobe.cpp (FEATURE_STACK_PROBE)? Doesn't seem to be used.

**Comment by jkotas** [comment]
`FEATURE_STACK_PROBE` has been always off for CoreCLR.

**Comment by mikedn** [comment]
OK, remove genericstackprobe.cpp as well. I've left in the following:

* vm\interpreter.cpp (maybe someone wants to play with it)
* vm\mdassistants.cpp (don't know why it's disabled, seems like a useful feature to have)
* vm\prestub.cpp (somewhat convoluted condition)
* vm\prof*.cpp (AFAIK profiler support will be enabled at some point)
* vm\umthunkhash.cpp (C++/CLI I presume, unlikely to see this in CoreCLR but I know people asked about it)
* vm\typeequivalencehash.cpp (COM PIA?)
* vm\gccover.cpp, dbggcinfoencoder.cpp, perfmap.cpp (debugging stuff, presumably useful to have around)
* utilcode\perflog.cpp (debugging stuff enabled only for 32 bit builds, unclear what stops it from being enabled on 64 bit builds as well)

and disabled LNK4221 for static libraries (it was wrongly disabled for dlls).

Fixes #2252

**Comment by mikedn** [comment]
Removed umthunkhash.cpp/typeequivalencehash.cpp and squashed. I also rebased against master, maybe the build will stop failing after you disabled those tests.

**Pull request merged**